### PR TITLE
Profiler runs without TF & TF only executes when necessary

### PR DESCRIPTION
**Pull request by lettergram**
PR makes it so you can load the profiler without loading tensorflow:

**If TensorFlow is installed**:
```python
>>> import dataprofiler as dp
>>> data = dp.Data('dataprofiler/tests/data/csv/guns.csv')
>>> profile = dp.Profiler(data) # TensorFlow loads here
2021-03-03 02:46:53.359832: I tensorflow/stream_executor/platform/....
```

**If TensorFlow is NOT installed**:
```python
>>> import dataprofiler as dp
>>> data = dp.Data('dataprofiler/tests/data/csv/guns.csv')
>>> profile = dp.Profiler(data)

!!! WARNING Partial Profiler Failure !!!

Profiling Type: data_labeler
Exception: AttributeError
Message: module 'tensorflow' has no attribute '__version__'

For labeler errors, try installing tensorflow and tensorflow-addons; command:
  $ pip install tensorflow tensorflow-addons --user

  self._create_profile(df_series, options)
```

-------------

This is related to issue #39, next steps (if we go that route):

1. Removing the requirement(s)
2. Move labeler requirements to `requirements-labeler.txt`
3. Create an "extras" in setup.py moving it to a special `dataprofiler[labeler]`

### CODE REVIEW

**Review by ChrisWallace2020 — DISMISSED** [review]

File: dataprofiler/labelers/character_level_cnn_model.py [hunk 1]
```diff
@@ -6,17 +6,27 @@
...
-import tensorflow as tf
-import tensorflow_addons as tfa
 import numpy as np
 from sklearn import decomposition
 
 from . import labeler_utils
 from .base_model import BaseModel, BaseTrainableModel
 from .base_model import AutoSubRegistrationMeta
 
+import tensorflow as tf
```
Comment (JGSweets) [inline dataprofiler/labelers/character_level_cnn_model.py:16]: needs to move back for pep8

File: dataprofiler/profilers/profile_builder.py [hunk 2]
```diff
@@ -329,6 +329,23 @@ def __init__(self, data, samples_per_update=None, min_true_samples=None,
         elif not isinstance(profiler_options, ProfilerOptions):
             raise ValueError("The profile options must be passed as a "
                              "ProfileOptions object.")
+
+        try:
```
Comment (JGSweets) [inline dataprofiler/profilers/profile_builder.py:333]: I'm not sure turning off the data labeler in general is appropriate as a labeler maybe made without TF. Need to think more on this of how best to address.

Comment (lettergram) [reply dataprofiler/profilers/profile_builder.py:333]: I actually don't know where else to put this, but currently there's no other reasonable model besides tensorflow. 

I agree this is very convoluted, but I don't know where else to put this.

Comment (gme5078) [reply dataprofiler/profilers/profile_builder.py:333]: I agree with Jeremy, a labeler could be used without tensorflow (like our own regex model). Going to look in the repo for where I would put this.

**Review by gme5078 — CHANGES_REQUESTED** [review]

File: dataprofiler/profilers/profile_builder.py [hunk 3]
```diff
@@ -329,6 +329,23 @@ def __init__(self, data, samples_per_update=None, min_true_samples=None,
...
+            import tensorflow_addons
+        except Exception:
+            import warnings
+            warning_msg = "\n\nTensorFlow and/or TensorFlow Addons"
+            warning_msg += "are not installed.\n\nUnfortunately, this "
+            warning_msg += "means the library cannot label the data\n\n"
+            warning_msg += "To label the data, install tensorflow & "
+            warning_msg += "tensorflow-addons\n\n"
+            warning_msg += "    $ pip install tensorflow "
+            warning_msg += "tesnorflow-addons --user\n\n"
```
Comment (gme5078) [inline dataprofiler/profilers/profile_builder.py:344]: typo

**Review by gme5078 — CHANGES_REQUESTED** [review]

**Review by AnhTruong — DISMISSED** [review]

File: dataprofiler/profilers/column_profile_compilers.py [hunk 4]
```diff
@@ -43,20 +43,41 @@ def _create_profile(self, df_series, options=None):
...
+                    # This is considered a major error
+                    if type(e).__name__ == "ValueError":
+                        raise ValueError(e)
+                    
+                    warning_msg += "\n\nFor labeler errors, try installing "
+                    warning_msg += "tensorflow and tensorflow-addons; command:\n"
+                    warning_msg += "  $ pip install tensorflow "
+                    warning_msg += "tensorflow-addons --user\n"
+
+                    warnings.warn(warning_msg, RuntimeWarning, stacklevel=2)
```
Comment (gme5078) [inline dataprofiler/profilers/column_profile_compilers.py:79]: Does this exception occur for every column of the dataset?
What doess warning_flag = False do?

**Review by gme5078 — CHANGES_REQUESTED** [review]

File: dataprofiler/profilers/column_profile_compilers.py (hunk 4 above)
Comment (lettergram) [reply dataprofiler/profilers/column_profile_compilers.py:78]: Yes, if it fails on every potential issue

File: .gitignore [hunk 5]
```diff
@@ -3,6 +3,7 @@
 DataProfiler\.egg-info/
 DataProfiler.egg-info/*
 data_profiler/labelers/embeddings/glove-reduced-64D.txt
```
Comment (gme5078) [inline .gitignore:5]: feels like this one should be removed, but not worth another commit

**Review by AnhTruong — APPROVED** [review]

**Review by gme5078 — APPROVED** [review]

**Pull request merged**